types/navigation-react-native.d.ts

```diff
@@ -52,6 +52,5375 @@ type Animation = TranslateAnimation | ScaleAnimation | AlphaAnimation | RotateAn
 
 type Transition = { type: 'sharedAxis', axis?: 'x' | 'y' | 'z' } | { type: 'elevationScale' | 'fade' | 'fadeThrough' | 'hold' } | Animation | Animation[] | { duration?: number, items: Animation[] };
 
+type SFSymbols = 
```

Comment:
@grahammendick do you think I should add them to a separate file? Since it's huge.

Reply:
Wow, that’s clever but a bit scary. Might be a bit less scary if it was a separate file. I think it’s very good but I might need to think about it for a bit. What do you think about it?

How did you get the full list? The list will change whenever a new version of iOS comes out, right? I guess it doesn’t really matter if the list isn’t completely up to date because typescript won’t complain?

Reply:
I think it should be called `SFSymbol` instead of `SFSymbols` (singular not plural)

Reply:
Also not sure it should extend `ImageURISource` because we only currently support `uri` property. Shouldn't it be this?
```ts
image?: ImageRequireSource | ImageURISource | { uri: SFSymbol };
```

Reply:
> What do you think about it?

I really want it to be on a separate file. I just left it there initially just for testing, but personally I believe it should belong to it's own file and import it here so we keep things clean.

> How did you get the full list? The list will change whenever a new version of iOS comes out, right? 

I believe so, yes. The full list comes from the SF Symbols app for macOS, you just Cmd + A the entire list then go to edit and copy the names, pretty easy to get the entire thing :)

<img width="507" alt="image" src="https://github.com/grahammendick/navigation/assets/36211892/e2f85ca7-ea31-486d-a3b8-a5b4c2437c71">

> I guess it doesn’t really matter if the list isn’t completely up to date because typescript won’t complain?

Yep, we can update it on their next major release if we want to, no need to keep an active eye on it

Reply:
> Here I combined chart.line.uptrend.xyaxis with gearshape.fill, using gearshape.fill as a badge.

You said you could [combine symbols together](https://github.com/grahammendick/navigation/pull/777#issuecomment-1987429491). Is that something we can capture in typescript maybe? How do we know which ones combine? I guess it's unlikely anybody would want to do that for the tab bar or bar button?

Reply:
If it's not a pain, can you use single quote instead of double please? so 'square.and.arrow.up' instead of "square.and.arrow.up"

Reply:
> Also not sure it should extend ImageURISource because we only currently support uri property. Shouldn't it be this?

Yeah, you're right. Maybe in the future it could have its own `SFSymbolURISource` type to send [configuration](https://developer.apple.com/documentation/uikit/uiimage/symbolconfiguration) or adding [symbol effects](https://developer.apple.com/documentation/uikit/uiimageview/4210356-addsymboleffect)

> You said you could [combine symbols together](https://github.com/grahammendick/navigation/pull/777#issuecomment-1987429491). Is that something we can capture in typescript maybe?

Oh yeah, that's something I found being curious around with them from the SF Symbols app, I think in order to use them you (have to add them into the asset catalog)[https://developer.apple.com/documentation/uikit/uiimage/creating_custom_symbol_images_for_your_app#3875966], so I don't see true benefit for them. 

> If it's not a pain, can you use single quote instead of double please? so 'square.and.arrow.up' instead of "square.and.arrow.up"

Sure!

Reply:
> But making a TypeScript definition for them is quite easy, if you want I can make them,

No, don't type combined symbols. I can't think of a use for them for tab bar or bar button

Reply:
Forgot to mention, you can [use navigation-react-native-tests.tsx to test your types](https://github.com/grahammendick/navigation/blob/master/NavigationReactNative/test/navigation-react-native-tests.tsx) if you want